src/js/pouchPersisted.js

```diff
@@ -0,0 +1,253 @@
+    // some further assumptions about desired behaviour, including:
+    // - a create/update pattern when doing set operations
+    // - a "get before set/delete" pattern that automatically handles revisions
+    // - the automatic generation of timestamp-based IDs for persisted models
+    fluid.defaults("floe.dashboard.pouchPersisted", {
+        gradeNames: ["floe.dashboard.eventInTimeAware"],
+        components: {
+            dataSource: {
+                type: "floe.dashboard.dataSource.pouchDB",
+                options: {
+                    dbOptions: "{pouchPersisted}.options.dbOptions"
+                }
+            }
+        },
+        events: {
+            "onSetPouchId": null,
+            // Event signature is the retrieved doc
+            "onPouchDocRetrieved": null,
+            // Event signature is the set request response
+            "onPouchDocStored": null,
+            // Event signature is the delete request response
+            "onPouchDocDeleted": null,
+            // Event signature for errors should include the error structure
+            // returned by floe.dashboard.pouchPersisted.makeErrorStructure
+            // (a basic message + the Pouch error structure)
+            "onPouchGetError": null,
+            "onPouchSetError": null,
+            "onPouchDeleteError": null
+        },
+        listeners: {
+            "onCreate.setPouchId": {
+                funcName: "{that}.setPouchId",
+                args: "{that}"
+            }
+        },
+        dbOptions: {
+            // name: "test",
+        },
+        // Implementing grades can use these invokers in combinations with
+        // events and listeners to store, retrieve and delete model as
+        // appropriate
+        invokers: {
+            // Implementing grades or instances that want a different ID
+            // strategy should override this invoker
+            "setPouchId": {
+                funcName: "floe.dashboard.pouchPersisted.setPouchIdToCurrentTime",
+                args: "{that}"
+            },
+            "get": {
+                funcName: "floe.dashboard.pouchPersisted.get",
+                args: ["{that}", "{arguments}.0"]
+            },
+            "set": {
+                funcName: "floe.dashboard.pouchPersisted.set",
+                args: "{that}"
+            },
+            "delete": {
+                funcName: "floe.dashboard.pouchPersisted.del",
+                args: "{that}"
+            },
+            "persist": {
+                funcName: "floe.dashboard.pouchPersisted.persist",
+                args: "{that}"
+            }
+        }
+    });
+
+    // Sets the ID to to "created" timestamp to make filtering
+    // by time easier
+    floe.dashboard.pouchPersisted.setPouchIdToCurrentTime = function (that) {
+        that.applier.change("_id", that.model.timeEvents.created);
+        that.events.onSetPouchId.fire();
```

Comment:
This event can be removed in favour of advertising that the model change event can be used